[BUG]: Can't compile with Achievements disabled

### Describe the Bug

If i turn off the USE_ACHIEVEMENTS option the compilation fails

### Reproduction Steps

set USE_ACHIEVEMENTS to OFF

### Expected Behavior

for it to be compiled with USE_ACHIEVEMENTS set to OFF

### PCSX2 Revision

1.7.4263

### Operating System

Linux (64bit) - Specify distro below

### If Linux - Specify Distro

Artix

Tried to build without achievements to verify and get an error.

Here the terminal output : 

> /home/{user}/Documents/Compilation/PCSX2/pcsx2-1.7.4263/pcsx2/Frontend/CommonHotkeys.cpp: Dans la fonction « void HotkeyAdjustTargetSpeed(double) »:
/home/{user}/Documents/Compilation/PCSX2/pcsx2-1.7.4263/pcsx2/Frontend/CommonHotkeys.cpp:46:34: erreur: « Achievements » n'a pas été déclaré
   46 |         const double min_speed = Achievements::ChallengeModeActive() ? 1.0 : 0.1;
      |                                  ^~~~~~~~~~~~
make[2]: *** [pcsx2/CMakeFiles/PCSX2.dir/build.make:4365: pcsx2/CMakeFiles/PCSX2.dir/Frontend/CommonHotkeys.cpp.o] Error 1
make[2]: *** Attente des tâches non terminées....
/home/{user}/Documents/Compilation/PCSX2/pcsx2-1.7.4263/pcsx2/Frontend/FullscreenUI.cpp:6962:43: erreur: « bool FullscreenUI::OpenAchievementsWindow() » aurait dû être déclaré à l'intérieur de « FullscreenUI »
 6962 | bool FullscreenUI::OpenAchievementsWindow()
      |                                           ^
/home/{user}/Documents/Compilation/PCSX2/pcsx2-1.7.4263/pcsx2/Frontend/FullscreenUI.cpp:6962:6: erreur: rend ambigüe la nouvelle déclaration de « bool FullscreenUI::OpenAchievementsWindow() »
 6962 | bool FullscreenUI::OpenAchievementsWindow()
      |      ^~~~~~~~~~~~
Dans le fichier inclus depuis /home/{user}/Documents/Compilation/PCSX2/pcsx2-1.7.4263/pcsx2/Frontend/FullscreenUI.cpp:21:
/home/{user}/Documents/Compilation/PCSX2/pcsx2-1.7.4263/pcsx2/./Frontend/FullscreenUI.h:36:14: note: ancienne déclaration « void FullscreenUI::OpenAchievementsWindow() »
   36 |         void OpenAchievementsWindow();
      |              ^~~~~~~~~~~~~~~~~~~~~~
/home/{user}/Documents/Compilation/PCSX2/pcsx2-1.7.4263/pcsx2/Frontend/FullscreenUI.cpp:6967:43: erreur: « bool FullscreenUI::OpenLeaderboardsWindow() » aurait dû être déclaré à l'intérieur de « FullscreenUI »
 6967 | bool FullscreenUI::OpenLeaderboardsWindow()
      |                                           ^
/home/{user}/Documents/Compilation/PCSX2/pcsx2-1.7.4263/pcsx2/Frontend/FullscreenUI.cpp:6967:6: erreur: rend ambigüe la nouvelle déclaration de « bool FullscreenUI::OpenLeaderboardsWindow() »
 6967 | bool FullscreenUI::OpenLeaderboardsWindow()
      |      ^~~~~~~~~~~~
/home/{user}/Documents/Compilation/PCSX2/pcsx2-1.7.4263/pcsx2/./Frontend/FullscreenUI.h:37:14: note: ancienne déclaration « void FullscreenUI::OpenLeaderboardsWindow() »
   37 |         void OpenLeaderboardsWindow();
      |              ^~~~~~~~~~~~~~~~~~~~~~
/home/{user}/Documents/Compilation/PCSX2/pcsx2-1.7.4263/pcsx2/Frontend/FullscreenUI.cpp:6972:49: erreur: « void FullscreenUI::DrawAchievementsSettingsPage() » aurait dû être déclaré à l'intérieur de « FullscreenUI »
 6972 | void FullscreenUI::DrawAchievementsSettingsPage()
      |                                                 ^
make[2]: *** [pcsx2/CMakeFiles/PCSX2.dir/build.make:4381: pcsx2/CMakeFiles/PCSX2.dir/Frontend/FullscreenUI.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1071: pcsx2/CMakeFiles/PCSX2.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

--- a/pcsx2/Frontend/CommonHotkeys.cpp
+++ b/pcsx2/Frontend/CommonHotkeys.cpp
@@ -17,6 +17,7 @@
 #include "common/Assertions.h"
 #include "common/FileSystem.h"
 #include "common/Path.h"
+#include "pcsx2/Achievements.h"
 #include "Frontend/CommonHost.h"
 #include "Frontend/FullscreenUI.h"
 #include "Frontend/InputManager.h"

--- a/pcsx2/Frontend/FullscreenUI.cpp
+++ b/pcsx2/Frontend/FullscreenUI.cpp
@@ -4093,18 +4093,6 @@ void FullscreenUI::DrawFoldersSettingsPage()
 	EndMenuButtons();
 }
 
-#ifndef ENABLE_ACHIEVEMENTS
-
-void FullscreenUI::DrawAchievementsSettingsPage(std::unique_lock<std::mutex>& settings_lock)
-{
-	BeginMenuButtons();
-	ActiveButton(ICON_FA_BAN " This build was not compiled with Achievements support.", false, false,
-		ImGuiFullscreen::LAYOUT_MENU_BUTTON_HEIGHT_NO_SUMMARY);
-	EndMenuButtons();
-}
-
-#endif
-
 void FullscreenUI::DrawAdvancedSettingsPage()
 {
 	static constexpr const char* ee_rounding_mode_settings[] = {"Nearest", "Negative", "Positive", "Chop/Zero (Default)"};
@@ -6959,17 +6947,15 @@ void FullscreenUI::DrawAchievementsLoginWindow()
 
 #else
 
-bool FullscreenUI::OpenAchievementsWindow()
+void FullscreenUI::OpenAchievementsWindow()
 {
-	return false;
 }
 
-bool FullscreenUI::OpenLeaderboardsWindow()
+void FullscreenUI::OpenLeaderboardsWindow()
 {
-	return false;
 }
 
-void FullscreenUI::DrawAchievementsSettingsPage()
+void FullscreenUI::DrawAchievementsSettingsPage(std::unique_lock<std::mutex>& settings_lock)
 {
 	BeginMenuButtons();
 	ActiveButton(ICON_FA_BAN "  This build was not compiled with RetroAchievements support.", false, false,

